torch/autograd/function.py

```diff
@@ -585,6 +585,19 @@ def _compiled_autograd_key(ctx):
         return (ctx._autograd_function_id,)
 
 
+def is_setup_context_defined(fn):
+    return fn != _SingleLevelFunction.setup_context
+
+
+def autograd_function_forward_rewritten(original_forward, original_setup_context):
+    def new_forward(ctx, *args, **kwargs):
+        output = original_forward(*args, **kwargs)
+        original_setup_context(ctx, args, output)
+        return output
+
+    return new_forward
```

Comment:
Can we move this somewhere else (e.g. higher_order_ops.py)?

Reply:
This function should be in the ```MOD_INLINELIST``` to pass ```trace_rules.check```, so we can't put it in ```higher_order_ops.py```. Let me try to add it to ```torch/_functorch/autograd_function```. If it breaks other places, probably I'll go with a _ prefix.